### PR TITLE
tiflash: refine TiFlash configuration parameters

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

* Add more details about tiflash && tiflash-proxy logging config.
* Remove `raftstore.store-batch-retry-recv-timeout` from tiflash-proxy config. This item have been removed from tiflash-proxy since v6.5

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18099, https://github.com/pingcap/docs-cn/pull/18138
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -271,9 +274,29 @@ delta_index_cache_size = 0
...
+    ## If this parameter is not set or set to the default value `0`, TiFlash Proxy saves all log files.
+    ## If this parameter is set to a non-zero value, TiFlash Proxy retains at most the number of old log files specified by `max-backups`. For example, if you set it to `7`, TiKV retains at most 7 old log files.
+    max-backups = 0
+    ## The maximum number of days that the log files are retained. Introduced in v5.4.0.
+    ## If this parameter is not set or set to the default value `0`, TiFlash Proxy retains all log files.
+    ## If this parameter is set to a non-zero value, TiKV cleans up outdated log files after the number of days specified by `max-days`.
+    max-days = 0
+
 [server]
     engine-addr = The external access address of the TiFlash coprocessor service.
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:298]: These two lines are removed. We suggest user set it in the tiflash.toml (line 169~170)

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -271,9 +274,29 @@ delta_index_cache_size = 0
 
 ### Configure the `tiflash-learner.toml` file
 
+The parameters in `tiflash-learner.toml` are basically the same as those in TiKV. You can refer to [TiKV configuration](/tikv-configuration-file.md) for TiFlash configuration. The following are only commonly used parameters. Note that:
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:277]:
```suggestion
The parameters in `tiflash-learner.toml` are basically the same as those in TiKV. You can refer to [TiKV configuration](/tikv-configuration-file.md) for TiFlash proxy configuration. The following are only commonly used parameters. Note that:
```

File: tiflash/tiflash-configuration.md [hunk 3]
```diff
@@ -271,9 +274,29 @@ delta_index_cache_size = 0
...
+
 ```toml
+[log]
+    ## The log level of TiFlash proxy (available options: "trace", "debug", "info", "warn", "error"). The default value is "info". Introduced in v5.4.0.
+    level = "info"
+
+[log.file]
+    ## The maximum number of log files to save. Introduced in v5.4.0.
+    ## If this parameter is not set or set to the default value `0`, TiFlash Proxy saves all log files.
+    ## If this parameter is set to a non-zero value, TiFlash Proxy retains at most the number of old log files specified by `max-backups`. For example, if you set it to `7`, TiKV retains at most 7 old log files.
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:290]:
```suggestion
    ## If this parameter is set to a non-zero value, TiFlash Proxy retains at most the number of old log files specified by `max-backups`. For example, if you set it to `7`, TiFlash Proxy retains at most 7 old log files.
```

File: tiflash/tiflash-configuration.md [hunk 4]
```diff
@@ -271,9 +274,29 @@ delta_index_cache_size = 0
...
+    max-backups = 0
+    ## The maximum number of days that the log files are retained. Introduced in v5.4.0.
+    ## If this parameter is not set or set to the default value `0`, TiFlash Proxy retains all log files.
+    ## If this parameter is set to a non-zero value, TiKV cleans up outdated log files after the number of days specified by `max-days`.
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:294]:
```suggestion
    ## If this parameter is set to a non-zero value, TiFlash Proxy cleans up outdated log files after the number of days specified by `max-days`.
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM

File: tiflash/tiflash-configuration.md [hunk 5]
```diff
@@ -271,9 +274,26 @@ delta_index_cache_size = 0
 
 ### Configure the `tiflash-learner.toml` file
 
+The parameters in `tiflash-learner.toml` are basically the same as those in TiKV. You can refer to [TiKV configuration](/tikv-configuration-file.md) for TiFlash Proxy configuration. The following are only commonly used parameters. Note that:
+
+1. Compared with TiKV, TiFlash Proxy has an extra `raftstore.snap-handle-pool-size` parameter.
+2. The `label` whose key is `engine` is reserved and cannot be configured manually.
```
Comment (lilin90) [inline tiflash/tiflash-configuration.md:280]:
```suggestion
- Compared with TiKV, TiFlash Proxy has an extra `raftstore.snap-handle-pool-size` parameter.
- The `label` whose key is `engine` is reserved and cannot be configured manually.
```

**Review by Oreoxmt — APPROVED** [review]

**Review by lilin90 — COMMENTED** [review]
/approve

**Pull request merged**